markdown.go

```diff
@@ -314,24 +319,20 @@ func MarkdownBasic(input []byte) []byte {
 //
 // * Custom Heading IDs
 func MarkdownCommon(input []byte) []byte {
-	// set up the HTML renderer
-	renderer := NewHTMLRenderer(HTMLRendererParameters{
-		Flags: CommonHTMLFlags,
-	})
-	return Markdown(input, renderer, DefaultOptions)
+	return Markdown(input, DefaultOptions)
 }
 
 // Markdown is the main rendering function.
 // It parses and renders a block of markdown-encoded text.
-// The supplied Renderer is used to format the output, and extensions dictates
-// which non-standard extensions are enabled.
+// The supplied options contain a Renderer used to format the output, and
+// extensions that dictate which non-standard extensions are enabled.
 //
-// To use the supplied HTML renderer, see NewHTMLRenderer.
-func Markdown(input []byte, renderer Renderer, options Options) []byte {
-	if renderer == nil {
-		return nil
+// If options.Renderer is nil, the supplied HTML renderer is used.
```

Comment:
> If options.Renderer is nil, the supplied HTML renderer is used.

That's not accurate. I think you meant "If options.Renderer is nil, the default HTML renderer is used."

I would even consider writing it more explicitly as:

```
// If options.Renderer is nil, then DefaultOptions.Renderer is used.
```

That's equally readable, more accurate, and takes care of the situation if the user modifies `DefaultOptions`. Should we document that `DefaultOptions` should not be modified, or at least warn that if it is, that has effect on other methods?